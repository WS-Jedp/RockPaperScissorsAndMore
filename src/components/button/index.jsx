import React from 'react'
import { ButtonSecondaryWrapper, ButtonPrimaryWrapper } from './styles'

export const ButtonSecondary = ({title = 'Rules', action = () => {}}) => (
  <ButtonSecondaryWrapper onClick={action}>
    {title}
  </ButtonSecondaryWrapper>
)

export const ButtonPrimary = ({title = 'Play Again', action = () => {}}) => (
  <ButtonPrimaryWrapper onClick={action}>
    {title}
  </ButtonPrimaryWrapper>
)
