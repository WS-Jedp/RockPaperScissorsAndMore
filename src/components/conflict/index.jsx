import React from 'react'
import { ConflictWrapper } from './styles'

import { PickedOption } from '../pickedOption'
import { PlayAgain } from '../playAgain'

export const Conflict = ({userOption = {}, homeOption = {}}) => {

  const isMobile = false
  
  return (
    <>
      <ConflictWrapper>
        <PickedOption title='You Picked' option={userOption.name} color={userOption.color} picked={true} />
        {
        isMobile && <PlayAgain title='You Win'action={() => {}} />
        }
        <PickedOption title='The House Picked' option={homeOption.name} color={homeOption.color} />
      </ConflictWrapper>
      {
        !isMobile && <PlayAgain title='You Win'action={() => {}} />
      }
    </>
  )
}
