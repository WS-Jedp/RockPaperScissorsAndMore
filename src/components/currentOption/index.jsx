import React from 'react'
import { IconLizard, IconPaper, IconScissors, IconRock, IconSpock } from '../icons'

export const CurrentOption = ({option = 'rock'}) => {

  const optionName = option.toLowerCase(); 

    switch (optionName) {
      case "rock":
        return <IconRock />
        break;
      case "paper":
        return <IconPaper />
        break;
      case "scissors":
    return <IconScissors />
        break;
      case "lizard":
        return <IconLizard />
        break;
      case "spock":
        return <IconSpock />
        break;
    
      default:
        return <IconRock />
        break;
    }
  }