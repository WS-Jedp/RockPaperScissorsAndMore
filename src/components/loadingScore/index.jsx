import React from 'react'
import { LoadingWrapper } from './styles'

export const LoadingScore = () => (
  <LoadingWrapper />
)
