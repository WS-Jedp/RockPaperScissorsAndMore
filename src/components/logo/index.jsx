import React from 'react'
import logoSVG from '../../assets/images/logo-bonus.svg'

import { LogoWrapper } from './styles'

export const Logo = () => (
  <LogoWrapper aria-label="Logo Rock, Paper, Scissors And More">
    <img src={logoSVG} alt="Logo Rock, Paper, Scissors And More"/>
  </LogoWrapper>
)