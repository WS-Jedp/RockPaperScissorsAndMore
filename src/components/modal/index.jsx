import React from 'react'
import ReactDOM from 'react-dom'
import { ModalWrapper } from './styles'

export const Modal = ({children}) => ReactDOM.createPortal((
  <ModalWrapper>
    { children || 'Hello World' }
  </ModalWrapper>), 
  window.document.getElementById('modal-portal'))
