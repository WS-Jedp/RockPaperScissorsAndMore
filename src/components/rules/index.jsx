import React from 'react'
import { RulesWrapper, HeaderRules, ImageRules } from './styles'

import { IconClose } from '../icons'

export const Rules = ({title, img}) => {
  return (
    <RulesWrapper>
      <HeaderRules>
        <h2>{title}</h2>
        <div className="close-rules-up">
          <IconClose />
        </div>
      </HeaderRules>

      <ImageRules aria-label="Rules of the game">
        <img src={img} alt="Rules of the game"/>
      </ImageRules>
      <div className="close-rules-down">
        <IconClose width="30" height="30" />
      </div>
    </RulesWrapper>
  )
}
