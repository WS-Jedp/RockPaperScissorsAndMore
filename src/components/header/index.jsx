import React from 'react'
import { WrapperHeader } from './styles'

import { Score } from '../score/index'
import { Logo } from '../logo'

export const Header = () => (
  <WrapperHeader>
    <Logo />
    <Score />
  </WrapperHeader>
) 