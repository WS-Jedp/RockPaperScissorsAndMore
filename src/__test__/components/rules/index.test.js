import React from 'react'
import { mount } from 'enzyme'
import { create } from 'react-test-renderer'
import { Rules } from '../../../components/rules'

describe('Suite of test for the Rules component :D', () => {

  
  describe('Visual test for Rules components' , () => {
    const RulesComponent = create(<Rules />)
    test('should be the same visual component', () => {
      expect(RulesComponent.toJSON()).toMatchSnapshot()
    })

    test('should the title be "RULES"', () => {
      expect(mount(<Rules title="RULES" />).find('h2').text()).toBe('RULES')
    })
    
  })

})