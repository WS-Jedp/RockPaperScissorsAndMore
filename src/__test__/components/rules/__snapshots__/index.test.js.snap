// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite of test for the Rules component :D Visual test for Rules components should be the same visual component 1`] = `
<article
  className="sc-AxjAm iEPgwd"
>
  <div
    className="sc-AxiKw gpYzvK"
  >
    <h2 />
    <div
      className="close-rules-up"
    >
      <svg
        height={20}
        onClick={[Function]}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.97 0l2.122 2.121-7.425 7.425 7.425 7.425-2.121 2.12-7.425-7.424-7.425 7.425L0 16.97l7.425-7.425L0 2.121 2.121 0l7.425 7.425L16.971 0z"
          fill="#3B4262"
          fillRule="evenodd"
          opacity=".25"
        />
      </svg>
    </div>
  </div>
  <figure
    aria-label="Rules of the game"
    className="sc-AxirZ qzCQ"
  >
    <img
      alt="Rules of the game"
    />
  </figure>
  <div
    className="close-rules-down"
  >
    <svg
      height="30"
      onClick={[Function]}
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.97 0l2.122 2.121-7.425 7.425 7.425 7.425-2.121 2.12-7.425-7.424-7.425 7.425L0 16.97l7.425-7.425L0 2.121 2.121 0l7.425 7.425L16.971 0z"
        fill="#3B4262"
        fillRule="evenodd"
        opacity=".25"
      />
    </svg>
  </div>
</article>
`;
